### DISCUSSION TITLE
Mock Prisma client

### DISCUSSION THREAD

**Discussion by thermsdorff** [TypedProject/ts-express-decorators#2152] [Q&A] [answered]
Hello !

I need to mock prisma client to avoid database connection during tests.
Prisma recommend to mock client directly : https://www.prisma.io/docs/guides/testing/unit-testing

But since the instantiation of the Prisma client is handled by @tsed/prisma what is the best way to mock Prisma client while using tsed ?

**Comment by Romakita** [comment, marked as answer]
Hello @thermsdorff,

You can try that:

```typescript

import { PrismaClient } from '@prisma/client'
import { mockDeep, mockReset, DeepMockProxy } from 'jest-mock-extended'

export function mockPrismaService() {
   registerProvider({
      provide: PrismaService,
      useClass: null,
      useFactory: () => {
        return mockDeep<PrismaClient>()
      },
      hooks: {
        $onDestroy(instance: any) {
           mockReset(instance);
        }
      }
   })
}
export function restorePrismaService() {
     registerProvider({provide: PrismaService, useClass: PrismaService})
}
```

**Reply by thermsdorff** [reply]
Hello @Romakita !

Thank you for quick answer.

I tried to inject mock through PlatformTest invoke method, but it seems PrismaService onInit method is trigger before at platformTest init through injectorService.

Can you share me a way to override injectorService during tests ?

PS: I have successfully mock prisma repositories but still get stuck with prismaService initialisation

**Reply by Romakita** [reply]
Hello,
I already shared you a way to override injector.
You have to mockPrismaService before PlateformTest.bootstrap()

**Reply by thermsdorff** [reply]
Works like a charm ! 

I was using the mock after the bootstrap 🤦 

Thanks for your time

**Reply by Romakita** [reply]
Perfect ;)

**Reply by Romakita** [reply]
Hello @thermsdorff 

I decided to implement an official way to mock service directly on PlatformTest.create, PlatformTest.bootstrap!

The testing page on the website will describe official the process to mock service correctly ;).

Thanks for your discussion because this feature was missing!

See you
Romain
